### DISCUSSION TITLE
Mutiple layers and on-click events

### DISCUSSION THREAD

**Discussion by Harsha-Kasi** [visgl/deck.gl#7890] [Q&A] [answered]
Hello peeps!

I am new to Pydeck and I am trying to create multiple column layers with Pydeck, but only my first layer is being displayed. Any solutions to this on how to code multiple layers?

layer=[
    pdk.Layer(
                'ColumnLayer',
                data=df,
                get_position =["lon", "lat"],
                get_elevation ="total_weight",
                auto_highlight = True,
                radius = 3,
                elevation_scale = 0.5,
                pickable = True,
                extruded = True,
                get_Fill_Color = [48, 128, 255, 255],
            ),
    pdk.Layer(
                'ColumnLayer',
                data=df,
                get_position =["lon1", "lat1"],
                get_elevation ="total_volume",
                auto_highlight = True,
                radius = 3,
                elevation_scale = 0.5,
                pickable = True,
                extruded = True,
                get_Fill_Color = [48, 128, 128, 255],
            )
    ]
view_state = pdk.ViewState(
    longitude=df['lon'].mean(), latitude=df['lat'].mean(), zoom=15, min_zoom=3, max_zoom=20, pitch=30, bearing=10,
)

r = pdk.Deck(layers=[layer], initial_view_state=view_state)

Also, I would like to know if we can do on-click events. Please let me know.

**Comment by ibgreen** [comment, marked as answer]
Using different ids for layers is important in deck.gl. Especially when using multiple instances of the same layer type.
